scripts/typescript/extract.js

```diff
@@ -47,3 +41,24 @@ if (!result.succeeded) {
 
   process.exitCode = 1;
 }
+
+// move the right entry points
+// esm to override umd
+shell.mv(
+  path.join(__dirname, '../../es/index.es.d.ts'),
+  path.join(__dirname, '../../es/index.d.ts')
+);
+
+// production and development umd
+shell.cp(
+  path.join(__dirname, '../../dist/instantsearch.production.d.ts'),
+  path.join(__dirname, '../../dist/instantsearch.production.min.d.ts')
+);
+shell.cp(
+  path.join(__dirname, '../../dist/instantsearch.production.d.ts'),
+  path.join(__dirname, '../../dist/instantsearch.development.d.ts')
+);
+shell.cp(
+  path.join(__dirname, '../../dist/instantsearch.production.d.ts'),
+  path.join(__dirname, '../../dist/instantsearch.development.min.d.ts')
```

Comment:
in which case do people need development.d.ts file?

Reply:
all the files are the same, but typescript works by matching a .d.ts file to a .js file, so if you're importing /dist/development.min.js then the .d.ts is needed for that to work

Reply:
Thanks for explaining!